build/js/live-editor.output_pjs.js

```diff
@@ -1152,6 +1152,127 @@ var BabyHint = {
 // TODO(jlfwong): Stop globalizing BabyHint
 window.BabyHint = BabyHint;
 
+var PJSResourceCache = Backbone.Model.extend({
+
+    initialize: function(options) {
+        this.canvas = options.canvas;   // customized Processing instance
+        this.output = options.output;   // LiveEditorOutput instance
+        this.cache = {};
+        this.imageHolder = null;
+    },
+
+    // Load and cache all resources (images and sounds) that could be used in 
+    // the environment.  Right now all resources are loaded as we don't have 
+    // more details on exactly which images will be required.
+    //
+    // Execution is delayed once a getImage/getSound appears in the source code
+    // and none of the resources are cached. Execution begins once all the
+    // resources have loaded.
+    cacheResources: function(userCode, callback) {
+        var getImageRegex = /getImage\s*\(['"](.*?)['"]\)/g;
+        var images = this.getRegexMatches(userCode, getImageRegex, 1);
+        var getSoundRegex = /getSound\s*\(['"](.*?)['"]\)/g;
+        var sounds = this.getRegexMatches(userCode, getSoundRegex, 1);
```

Comment:
Originally I had separate methods called ```getSoundFilenames``` and ```getImageFilenames```... they we quite similar to so I decided to DRY the code out and ended up with this ```getRegexMatches``` method which I'm not sure is any better.